packages/capacitor/src/hooks/useOrientation.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+type Orientation = 'portrait' | 'landscape'
+
+function useOrientation() {
+  const [orientation, setOrientation] = useState<Orientation>(
+    window.innerHeight > window.innerWidth ? 'portrait' : 'landscape'
+  )
+
+  const portraitQuery = useMemo(
+    () => window.matchMedia('(orientation: portrait)'),
+    []
+  )
+
+  // Event Listeners
+  // ------------------------------------------------------------------------- /
+  const onOrientationChange = useCallback((event: MediaQueryListEvent) => {
+    setOrientation(event.matches ? 'portrait' : 'landscape')
+  }, [])
```

Comment:
Recommended to handle this in useEffect, where you don't need useMemo and useCallback.